WordPress/src/main/java/org/wordpress/android/ui/reader/utils/FeaturedImageUtils.kt

```diff
@@ -0,0 +1,35 @@
+package org.wordpress.android.ui.reader.utils
+
+import java.net.MalformedURLException
+import java.net.URL
+import javax.inject.Inject
+
+class FeaturedImageUtils
+@Inject constructor() {
+    fun showFeaturedImage(
+        featuredImage: String,
+        postText: String
+    ): Boolean {
+        return try {
+            val featuredImageUrl = URL(featuredImage)
+            val endIndex = featuredImageUrl.path.lastIndexOf("-", featuredImageUrl.path.lastIndexOf("/"))
+            val featuredImageFile = featuredImageUrl.file
+            val featuredImagePath = if (endIndex > 0 && !featuredImageFile.startsWith("-")) {
+                featuredImageUrl.path.substring(0, endIndex)
+            } else {
+                featuredImageUrl.path
+            }
+            when {
+                postText.contains(featuredImage) -> false
+                postText.contains(featuredImagePath) && featuredImagePath != featuredImageFile -> false
+                else -> true
+            }
+        } catch (e: MalformedURLException) {
+            false
+        }
+    }
+
+    companion object {
+        fun getInstance() = FeaturedImageUtils()
+    }
```

Comment:
I would consider to use the @JvmStatic annotation here so it can be accessed with a static-like syntax; wdyt?

Reply:
I'm replacing it with an injected call instead. I was using the `companion object` as a replacement of a static class but we shouldn't have that at all.

Reply:
Agree 👍!